horizons/__init__.py

```diff
@@ -0,0 +1,281 @@
+        delta_φ = dec2 - dec1
+        delta_λ = ra2 - ra1
+        a = np.sin(delta_φ / 2)**2 + np.cos(φ1) * np.cos(φ2) * np.sin(delta_λ / 2)**2
+        return 2 * np.arctan2(np.sqrt(a), np.sqrt(1 - a))
+
+    @staticmethod
+    def interpolate(ra1: float, dec1: float, ra2: float, dec2: float, f: float) -> float:
+        """
+        Interpolate between two coordinates.
+        based on https://github.com/gemini-hlsw/lucuma-core/blob/master/modules/core/shared/src/main/scala/lucuma/core/math/Coordinates.scala#L63-L89
+        """
+        # calculate angular distance to radians
+        delta = HorizonsClient.angular_distance(ra1, dec1, ra2, dec2)
+        if delta == 0:
+            return ra1, dec1
+        else:
+
+            φ1 = dec1
+            φ2 = dec2
+            λ1 = ra1
+            λ2 = ra2
+            a = np.sin((1 - f) * delta) / np.sin(delta)
+            b = np.sin(f * delta) / np.sin(delta)
+            x = a * np.cos(φ1) * np.cos(λ1) + b * np.cos(φ2) * np.cos(λ2)
+            y = a * np.cos(φ1) * np.sin(λ1) + b * np.cos(φ2) * np.sin(λ2)
+            z = a * np.sin(φ1) + b * np.sin(φ2)
+            φi = np.arctan2(z, np.sqrt(x * x + y * y))
+            λi = np.arctan2(y, x)
+            return λi, φi # there is a transformation here
+
+    def _form_horizons_name(self, tag: TargetTag, designation: str) -> str:
+        """
+        Formats the name of the body
+        """
+        if tag is TargetTag.COMET:
+            name = f'DES={designation};CAP'
+        elif tag is TargetTag.ASTEROID:
+            name = f'DES={designation};'
+        else:
+            name = self.generate_horizons_id(designation)
+        return name
+
+    def _time_bounds(self) -> Tuple[str, str]:
+        """
+        Returns the start and end times based on the given date
+        """
+        return self.start.strftime('%Y%m%d_%H%M'), self.end.strftime('%Y%m%d_%H%M')
+
+    def _get_ephemeris_file(self, name: str) -> str:
+        """
+        Returns the ephemeris file name
+        """
+        start, end = self._time_bounds()
+        return os.path.join(self.path, f"{self.site.name}_{name.replace(' ', '').replace('/', '')}_{start}-{end}.eph")
+    
+    def query(self,
+              target: str,
+              step: str = '1m',
+              make_ephem: str = 'YES',
+              cal_format: str = 'CAL',
+              quantities: str = '1',
+              object_data: str = 'NO',
+              daytime: bool = False,
+              csvformat: str = 'NO',
+              airmass: int = 3) -> requests.Response:
+     
+        # The items and order follow the JPL/Horizons batch example:
+        # ftp://ssd.jpl.nasa.gov/pub/ssd/horizons_batch_example.long
+        # and
+        # ftp://ssd.jpl.nasa.gov/pub/ssd/horizons-batch-interface.txt
+        # Note that spaces should be converted to '%20'
+        
+        skip_day = 'NO' if daytime else 'YES'
+        center = '568@399' if self.site is Site.GN else 'I11@399'
```

Comment:
I'm not familiar with this... are these specific to the sites? If so, I can move them to the mini-model where they should probably be and put them in `SiteInformation`. What do they represent?

Reply:
Those are specific to site: https://ssd.jpl.nasa.gov/horizons/manual.html#center

Reply:
I made a small PR to move these to the mini-model so that we can keep as much as the site information together so that the code is more easily adaptable to different observatories.

Once that's merged, you should just be able to use:
`self.site.coordinate_center`

Or reference it easier using:
```
center = self.site.coordinate_center
```
and then get rid of the `if` `else` on this line.